packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js

```diff
@@ -13,6 +13,30 @@ let ReactTestUtils;
 let SchedulerTracing;
 let act;
 let container;
+let isConcurrentMode;
+let concurrentRoot = null;
+let ReactFeatureFlags;
+
+function render(el, dom) {
+  if (isConcurrentMode) {
+    concurrentRoot = ReactDOM.unstable_createRoot(dom);
```

Comment:
I really dislike test abstractions like this. Especially when combined with mutable module state like `isConcurrentMode`. It makes writing tests easier, but it makes debugging failing tests harder.

If you do want to share code, I would pass `render` and `unmount` to a helper function that runs the tests cases, like we do in `ReactFiberNewContext-test`. Something like:

```js
runActTests(renderConcurrent, unmountConcurrent, 'concurrent');
runActTests(renderLegacySync, unmountLegacySync, 'legacy sync');
```